dali/python/nvidia/dali/external_source.py

```diff
@@ -155,81 +155,98 @@ class ExternalSource():
-    If specified, denotes the number of TensorLists produced by the source function
+    If specified, denotes the number of TensorLists that are produced by the source function.
 
 Keyword Args
 ------------
 
 `cycle`: bool
-    If ``True``, the source will be wrapped. Otherwise, StopIteration will be raised
-    when end of data is reached. This flag requires that ``source`` is either a collection, i.e. an
-    iterable object where ``iter(source)`` will return a fresh iterator on each call or a
-    generator function. In the latter case, the generator function will be called again when more
-    data is requested than was yielded by the function.
+    If set to ``True``, the source will be wrapped.
+
+    If set to ``False``, StopIteration is raised when the end of data is reached.
+    This flag requires that the ``source`` is a collection, for example, an iterative object where
+    ``iter(source)`` returns a fresh iterator on each call or a generator function.
+    In the latter case, the generator function is called again when more data than was
+    yielded by the function is requested.
 
 `name` : str, optional
-    The name of the data node - used when feeding the data in ``iter_setup``; can be omitted if
+    The name of the data node.
+
+    Used when feeding the data in ``iter_setup`` and can be omitted if
     the data is provided by ``source``.
 
+
 `layout` : :ref:`layout str<layout_str_doc>` or list/tuple thereof
-    If provided, sets the layout of the data. When ``num_outputs > 1``, layout can be a list
-    containing a distinct layout for each output. If the list has fewer elements than ``num_outputs``,
-    only the first outputs have the layout set, the reset have it cleared.
-
-`cuda_stream` : optional, `cudaStream_t` or an object convertible to `cudaStream_t`, e.g. `cupy.cuda.Stream`, `torch.cuda.Stream`
-    The CUDA stream, which is going to be used for copying data to GPU or from a GPU
-    source. If not set, best effort will be taken to maintain correctness - i.e. if the data
-    is provided as a tensor/array from a recognized library (CuPy, PyTorch), the library's
-    current stream is used. This should work in typical scenarios, but advanced use cases
-    (and code using unsupported libraries) may still need to supply the stream handle
-    explicitly.
-
-    Special values:
-      *  0 - use default CUDA stream
-      * -1 - use DALI's internal stream
-
-    If internal stream is used, the call to ``feed_input`` will block until the copy to internal
-    buffer is complete, since there's no way to synchronize with this stream to prevent
-    overwriting the array with new data in another stream.
+    If provided, sets the layout of the data.
+
+    When ``num_outputs > 1``, the layout can be a list that contains a distinct layout for each
+    output. If the list has fewer elements than the ``num_outputs`` value, only the first
+    outputs have the layout set, the rest of the outputs are cleared.
```

Comment:
```suggestion
    outputs have the layout set, the rest of the outputs don't have a layout set.
```

Reply:
That also reflects the recent change to layouts in external source - namely, that not providing one can keep the existing layout, if any, intact (which is relevant when using DALI's TensorListXPU as the source).

Reply:
Done